common/json-schema-to-grammar.cpp

```diff
@@ -78,30 +47,24 @@ struct BuiltinRule {
     std::vector<std::string> deps;
 };
 
-const std::string _up_to_15_digits = build_repetition("[0-9]", 0, 15);
-
 std::unordered_map<std::string, BuiltinRule> PRIMITIVE_RULES = {
     {"boolean", {"(\"true\" | \"false\") space", {}}},
-    {"decimal-part", {"[0-9] " + _up_to_15_digits, {}}},
-    {"integral-part", {"[0-9] | [1-9] " + _up_to_15_digits, {}}},
+    {"decimal-part", {"[0-9]{1,16}", {}}},
+    {"integral-part", {"[0-9] | [1-9] [0-9]{1,15}", {}}},
```

Comment:
Super minor nit, but woke up this morning with the thought that a better grammar might be:

`0 | [1-9] [0-9]{0,15}`

This way, a leading 0 is an "early exit" for the grammar, and anything else locks it into the other path.

Definitely not worth the brain cycles that I've spent on it, but I figured I might as well post about it anyways. :D